common/xfer/pipes.go

```diff
@@ -27,7 +27,16 @@ type pipe struct {
 	onClose         func()
 }
 
-// NewPipe makes a new... pipe.
+// NewPipeFromEnds makes a new pipe specifying its ends
+func NewPipeFromEnds(local io.ReadWriter, remote io.ReadWriter) Pipe {
```

Comment:
Only invocation of this has local set to `nil`.

I think I'd prefer to see a new type (that implements Pipe) but panics on `CopyToWebsocket` or something.